src/content-distribution/distribute/index.js

```diff
@@ -0,0 +1,209 @@
+					{ ! distribution.length ? (
+						<p>
+							{ __( 'This post has not been distributed to any connections yet.', 'newspack-network' ) }
+						</p>
+					) : (
+						<p>
+							{ sprintf(
+								_n(
+									'This post has been distributed with one connection.',
+									'This post has been distributed with %d connections.',
+									distribution.length,
+									'newspack-network'
+								),
+								distribution.length
+							) }
+						</p>
+					) }
+					<TextControl
+						placeholder={ __( 'Search available connections', 'newspack-network' ) }
+						value={ search }
+						disabled={ isSavingPost || isDistributing }
+						onChange={ setSearch }
+					/>
+				</PanelBody>
+				<PanelBody className="distribute-body">
+					{ selectableSites.length !== 0 && sites.length === networkSites.length && (
+						<CheckboxControl
+							name="select-all"
+							label={ __( 'Select all', 'newspack-network' ) }
+							disabled={ isSavingPost || isDistributing }
+							checked={ siteSelection.length === selectableSites.length }
+							indeterminate={ siteSelection.length > 0 && siteSelection.length < selectableSites.length }
+							onChange={ checked => {
+								setSiteSelection( checked ? selectableSites : [] );
+							} }
+						/>
+					) }
+					{ sites.map( siteUrl => (
+						<CheckboxControl
+							key={ siteUrl }
+							label={ getFormattedSite( siteUrl ) }
+							disabled={ distribution.includes( siteUrl ) || isSavingPost || isDistributing } // Do not allow undistributing a site.
+							checked={ siteSelection.includes( siteUrl ) || distribution.includes( siteUrl ) }
+							onChange={ checked => {
+								const urls = checked ? [ ...siteSelection, siteUrl ] : siteSelection.filter( url => siteUrl !== url );
+								setSiteSelection( urls );
+							} }
+						/>
+					) ) }
+				</PanelBody>
+				<PanelBody className="distribute-footer">
+					{ siteSelection.length > 0 && (
+						<p>
+							{ sprintf(
+								_n(
+									'One connection selected.',
+									'%d connections selected.',
+									siteSelection.length,
+									'newspack-network'
+								),
+								siteSelection.length
+							) }
+						</p>
+					) }
+					{ siteSelection.length > 0 && (
+						<Button
+							variant="secondary"
+							disabled={ isSavingPost || isDistributing }
+							onClick={ () => setSiteSelection( [] ) }
+						>
+							{ __( 'Clear', 'newspack-network' ) }
+						</Button>
+					) }
+					<Button
+						isBusy={ isDistributing }
+						variant="primary"
+						disabled={ siteSelection.length === 0 || isSavingPost || isDistributing }
+						onClick={ () => {
+							if ( hasChangedContent || isCleanNewPost ) {
+								savePost().then( distribute );
+							} else {
+								distribute();
+							}
+						} }
+					>
+						{ isDistributing ? __( 'Distributing...', 'newspack-network' ) : (
+							hasChangedContent || isCleanNewPost ?
+							__( 'Save & Distribute', 'newspack-network' ) :
```

Comment:
I was a little confused about the "Save and Distribute" button. It was unclear to me if that would publish the note or not. I realize there might not be room for "Save draft and distribute" on the button though 🤔